packages/core-bridge/src/runtime.rs

```diff
@@ -211,12 +211,18 @@ pub fn start_bridge_loop(channel: Arc<Channel>, receiver: &mut UnboundedReceiver
                     callback,
                 } => {
                     let client = (*client).clone();
-                    let worker = init_worker(config, client.into_inner());
-                    let (tx, rx) = unbounded_channel();
-                    tokio::spawn(start_worker_loop(worker, rx, channel.clone()));
-                    send_result(channel.clone(), callback, |cx| {
-                        Ok(cx.boxed(RefCell::new(Some(WorkerHandle { sender: tx }))))
-                    });
+                    match init_worker(&runtime, config, client.into_inner()) {
+                        Ok(worker) => {
+                            let (tx, rx) = unbounded_channel();
+                            tokio::spawn(start_worker_loop(worker, rx, channel.clone()));
```

Comment:
Probably not strictly necessary, but for consistency you might want to use `runtime.handle().spawn()` here